src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs

```diff
@@ -842,7 +842,9 @@ public void SecretMasker_PotentialSymmetricKeysAreClassified(string input, strin
     [DataTestMethod]
     [DataRow("ddddddddddddddddddddddddddddddddddddddddddddAzFu182vhA==", "SEC101/158.AzureFunctionIdentifiableKey:FF8E9A7C2A792029814C755C6704D9427F302E954DEF0FD5EE649BF9163E1F24")]
     [DataRow("eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee+ACRCTB7t/", "SEC101/176.AzureContainerRegistryIdentifiableKey:CE62C55A2D3C220DA0CBFE292B5A6839EC7F747C5B5A7A55A4E5D7D76F1C7D32")]
-    [DataRow("oy2mdeaddeaddeadeadqdeaddeadxxxezodeaddeadwxuq", "SEC101/031.NuGetApiKey:FC93CD537067C7F452073F24C7043D5F58E11B6F49546316BBE06BAA5747317E")]
+    [DataRow("oy2mdeaddeaddeadeadqdeaddeadxxxezodeaddeadwxuq", "SEC101/031.NuGetApiKey:FC93CD537067C7F452073F24C7043D5F58E11B6F49546316BBE06BAA5747317E")] // Fabricated input
+    [DataRow("oy2efu5r4dskzrpeslkab2aqhnwdyr4efhcnvvfx2qgkhm", "SEC101/031.NuGetApiKey:FC93CD537067C7F452073F24C7043D5F58E11B6F49546316BBE06BAA5747317E")] // V4, non-zero byte ID part, GUID-based password part
```

Comment:
>oy2efu5r4dskzrpeslkab2aqhnwdyr4efhcnvvfx2qgkhm [](http://example.com/codeflow?start=14&length=46)

So generally  we prefer not to encode data in source that precisely maps to generated key format.

A convention is to provider keys with repeated patterns that make it obvious we're not looking at randomized data. 

You can update the `GenerateTruePositiveExamples` method in the rule to produce a few of these.

These generated keys will flow to other automated tests that automatically perform testing, you don't need to add tests here. These tests are legacy hard-coded inlined tests from ADO that were put in place to ensure perfect compatibility with them before we switched to our package (that's actually in flight now).

oy11111111111111111111111111111111111111111
oyqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq

Reply:
Got it. I'll make this change.

Reply:
I removed these test changes and enhanced `GenerateTruePositiveExamples`.